addons/hr_timesheet/models/project.py

```diff
@@ -44,14 +46,35 @@ def _check_allow_timesheet(self):
 
     @api.depends('timesheet_ids')
     def _compute_total_timesheet_time(self):
+        timesheets_read_group = self.env['account.analytic.line'].read_group(
+            [('project_id', 'in', self.ids)],
+            ['project_id', 'unit_amount', 'product_uom_id'],
+            ['project_id', 'product_uom_id'],
+            lazy=False)
+        timesheet_time_dict = defaultdict(list)
+        uom_ids = set(self.timesheet_encode_uom_id.ids)
+
+        for result in timesheets_read_group:
+            uom_id = result['product_uom_id'] and result['product_uom_id'][0]
+            if uom_id:
+                uom_ids.add(uom_id)
+            timesheet_time_dict[result['project_id'][0]].append((uom_id, result['unit_amount']))
+
+        uoms_dict = {
+            uom['id']: {'factor': uom['factor'], 'factor_inv': uom['factor_inv']}
+            for uom in self.env['uom.uom'].search_read([('id', 'in', list(uom_ids))], ['id', 'factor', 'factor_inv'])
```

Comment:
I think the code will be easier to read if you user recordset instead of search read 
```suggestion
            uom.id: uom
            for uom in self.env['uom.uom'].browse(uom_ids)
```
You even save one query for the search